### ISSUE TITLE
Upgraded to 2.9.0 from 2.6.1 and now getting error errors:[/asc-stg-ham-tenant-01/Shared/ascendlearning_external_https_443__policy/rules/0/actions/0/select/pool/use: contains path to non-existent object

### ISSUE THREAD

**Issue by mitchellmaler** [F5Networks/k8s-bigip-ctlr#2481]
Setup Details
CIS Version : 2.9
Build: f5networks/k8s-bigip-ctlr:2.9
BIGIP Version: BIG-IP 15.1.5.1 Build 0.0.14 Point Release 1
AS3 Version: 3.36.1-1
Agent Mode: AS3
Orchestration: K8S
Orchestration Version: N/A
Pool Mode: Nodeport
Additional Setup details: N/A

#### Description  
Upgraded from 2.6 to 2.9 and CIS is now throwing this error:
```
2022/07/01 17:29:52 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/asc-stg-ham-tenant-01/Shared/ascendlearning_external_https_443__policy/rules/0/actions/0/select/pool/use: contains path to non-existent object istio_ingressgateway_external_443_istio_system] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.18.0/as3-schema-3.18.0-4.json"
```
  

#### Steps To Reproduce
1) CIS 2.6.1 deployed with multiple VirtualServers, TLSProfiles, etc. in a single namespace pointed to istio ingress gateway service
2) remove externaldnss CRD and update CRDs 
3) Upgrade to the latest helm chart and CIS

#### Expected Result 
CIS processes all existing VirtualServers with TLSProfiles and shows no change.

#### Actual Result  
Doesn't process anything due to the error above. The pool exists that the rule is pointed to so I am unsure why it says it is non-existent.

**Comment by mitchellmaler** [comment]
Had to revert back to 2.6.1, remove externaldnses in place for externaldnss, and edited the clusterrole to allow access to the old externaldnss as a temp fix until we can figure out why the upgrade doesn't work.

**Comment by mdditt2000** [comment]
@mitchellmaler what version of AS3 are you using?

**Comment by mitchellmaler** [comment]
@mdditt2000 Yes, I actually installed that version before the upgrade.
f5-appsvcs-3.36.1-1.noarch

```json
{
    "release": "1",
    "schemaCurrent": "3.36.0",
    "schemaMinimum": "3.0.0",
    "version": "3.36.1"
}
```

**Comment by mitchellmaler** [comment]
@trinaths @mdditt2000 Here are the rendered resources. There isn't anything sensitive as I create the creds secret outside of the chart. Everything seems to be pretty standard using as3, crd, node-ports, etc.

```yaml
---
# Source: f5-bigip-ctlr/templates/f5-bigip-ctlr-serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: release-name-f5-bigip-ctlr
  namespace: f5-bigip-ctlr-system
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: f5-bigip-ctlr
    app: f5-bigip-ctlr
    chart: f5-bigip-ctlr-0.0.20
    release: release-name
    heritage: Helm
---
# Source: f5-bigip-ctlr/templates/f5-bigip-ctlr-clusterrole.yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: release-name-f5-bigip-ctlr
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: f5-bigip-ctlr
    app: f5-bigip-ctlr
    chart: f5-bigip-ctlr-0.0.20
    release: release-name
    heritage: Helm
rules:
  - verbs:
      - get
      - list
      - watch
    apiGroups:
      - ''
      - apps
      - extensions
      - route.openshift.io
      - networking.k8s.io
    resources:
      - nodes
      - services
      - endpoints
      - namespaces
      - ingresses
      - ingressclasses
      - secrets
      - pods
      - routes
  - verbs:
      - get
      - list
      - watch
      - update
      - create
      - patch
    apiGroups:
      - ''
      - apps
      - extensions
      - route.openshift.io
      - networking.k8s.io
    resources:
      - configmaps
      - events
      - ingresses/status
      - routes/status
      - services/status
  - verbs:
      - get
      - list
      - watch
      - update
      - patch
    apiGroups:
      - cis.f5.com
    resources:
      - virtualservers
      - tlsprofiles
      - transportservers
      - externaldnses
      - ingresslinks
      - transportservers/status
      - virtualservers/status
      - ingresslinks/status
      - policies
---
# Source: f5-bigip-ctlr/templates/f5-bigip-ctlr-clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: release-name-f5-bigip-ctlr
  namespace: f5-bigip-ctlr-system
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: f5-bigip-ctlr
    app: f5-bigip-ctlr
    chart: f5-bigip-ctlr-0.0.20
    release: release-name
    heritage: Helm
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: release-name-f5-bigip-ctlr
subjects:
- kind: ServiceAccount
  name: release-name-f5-bigip-ctlr
  namespace: f5-bigip-ctlr-system
---
# Source: f5-bigip-ctlr/templates/f5-bigip-ctlr-deploy.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-f5-bigip-ctlr
  namespace: f5-bigip-ctlr-system
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: f5-bigip-ctlr
    app: f5-bigip-ctlr
    chart: f5-bigip-ctlr-0.0.20
    release: release-name
    heritage: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app: f5-bigip-ctlr
  template:
    metadata:
      labels:
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: f5-bigip-ctlr
        app: f5-bigip-ctlr
        release: release-name
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: app
                operator: In
                values:
                - f5-bigip-ctlr
            topologyKey: kubernetes.io/hostname
      serviceAccountName: release-name-f5-bigip-ctlr
      securityContext:
          runAsUser: 1000
          runAsGroup: 1000
          fsGroup: 1000
      containers:
      - name: f5-bigip-ctlr
        image: "container-proxy.ascendlearning.com/f5networks/k8s-bigip-ctlr:2.9.0"
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /health
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 15
          periodSeconds: 15
          successThreshold: 1
          timeoutSeconds: 15
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /health
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 30
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 15
        volumeMounts:
        - name: bigip-creds
          mountPath: "/tmp/creds"
          readOnly: true
        imagePullPolicy: Always
        command:
        - /app/bin/k8s-bigip-ctlr
        args:
        - --credentials-directory
        - /tmp/creds
        - --as3-validation=true
        - --bigip-partition=$PARTITION
        - --bigip-url=$SERVER
        - --custom-resource-mode=true
        - --disable-teems=true
        - --log-as3-response=true
        - --log-level=INFO
        - --namespace=istio-system
        - --node-label-selector=node-role.kubernetes.io/worker=true
        - --pool-member-type=nodeport
        resources:
          limits:
            cpu: 500m
            memory: 512Mi
          requests:
            cpu: 100m
            memory: 256Mi
      nodeSelector:
        node-role.kubernetes.io/control-plane: "true"
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/control-plane
        operator: Exists
      - effect: NoExecute
        key: node-role.kubernetes.io/etcd
        operator: Exists
      - effect: NoExecute
        key: CriticalAddonsOnly
        operator: Exists
      volumes:
      - name: bigip-creds
        secret:
           secretName: f5-bigip-ctlr-login
```

**Comment by nandakishorepeddi** [comment]
@mitchellmaler   I have tested with your configuration without irules(as those are custom irules). CIS is working as expected. Could you pls check my working configuration: https://github.com/nandakishorepeddi/2481 (Attached CIS log and AS3 declaration as well).  Observed that TLSProfile doesn't have host/hosts spec attribute and clientSSL needs a "/Common". Could you use the same as in above repo and let us know the status.

**Comment by mitchellmaler** [comment]
I updated the tls profiles with the leading / so it's /Common/etc and added the wildcard host to each tlsprofile. I then applied the deployment you had with just a few monifications (creds secret, security context, etc.). After doing that I am still seeing the same 422 error from the device. The old version worked without issue, not throwing that error.

I have emailed the full error from CIS. 

```
[ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/asc-stg-ham-tenant-01/Shared/ascendlearning_external_https_443__policy/rules/0/actions/0/select/pool/use: contains path to non-existent object istio_ingressgateway_external_443_istio_system] message:declaration is invalid]
``` 

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: test-bigip-controller-1
  name: test-bigip-controller-1
  namespace: f5-bigip-ctlr-system
spec:
  replicas: 1
  selector:
    matchLabels:
      app: test-bigip-controller-1
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: test-bigip-controller-1
    spec:
      containers:
      - args:
        - --credentials-directory
        - /tmp/creds
        - --bigip-partition
        - asc-stg-ham-tenant-01
        - --bigip-url
        - asc-stg-f504.ascendlearning.com
        - --log-level
        - DEBUG
        - --as3-validation=true
        - --disable-teems=true
        - --insecure
        - --log-as3-response=true
        - --custom-resource-mode=true
        - --pool-member-type
        - nodeport
        - --namespace
        - istio-system
        command:
        - /app/bin/k8s-bigip-ctlr
        image: f5networks/k8s-bigip-ctlr:2.9.0
        imagePullPolicy: Always
        name: test-bigip-controller-1
        volumeMounts:
          - mountPath: /tmp/creds
            name: bigip-creds
            readOnly: true
      securityContext:
        fsGroup: 1000
        runAsGroup: 1000
        runAsUser: 1000
      serviceAccount: f5-bigip-ctlr
      serviceAccountName: f5-bigip-ctlr
      volumes:
        - name: bigip-creds
          secret:
            defaultMode: 420
            secretName: f5-bigip-ctlr-login
```

**Comment by rajbaratht** [comment]
@nandakishorepeddi We are experiencing the same issue and waiting for a solution.
We have created a case with F5 #C3843508

Please keep us updated as well. Thanks!

**Comment by trinaths** [comment]
@mitchellmaler - Can you share the AS3 GET from this BIG-IP ?

**Comment by mitchellmaler** [comment]
@trinaths I will email the declare output for all tenants.

**Comment by mitchellmaler** [comment]
@trinaths I emailed it which I think @mdditt2000 receives. If you have not been able to get it please give me another address so I can send it to you.

**Comment by mdditt2000** [comment]
@trinaths check your inbox. thanks @mitchellmaler

**Comment by mitchellmaler** [comment]
@trinaths @mdditt2000 Is there any update on this? I can log a support case if that will help it along.

**Comment by trinaths** [comment]
@mitchellmaler Please log a support case. This will help with further investigation info.

**Comment by mitchellmaler** [comment]
@trinaths will do

**Comment by rajbaratht** [comment]
It's almost a month , is there any progress on this issue ? 
We upgraded to v2.9.1 and the issue persist.

**Comment by trinaths** [comment]
@rajbaratht - We tried to reproduce this issue in our lab and see no such issues. Is there any this specific VS yaml to test this further ?

**Comment by trinaths** [comment]
@mitchellmaler @rajbaratht  Can you share deployment and service definitions too ? I tried simple service and deployment with VS and TS profile above using fake irules. Unable to reproduce this issue.

**Comment by rajbaratht** [comment]
@trinaths I have attached all the details with the ticket #C3843508 and emailed you the logs as well. There are no changes to VS as they work fine in v2.8.1 and fails to work with either in v.2.9.0 or v2.9.1

**Comment by rajbaratht** [comment]
@trinaths Is the fix pushed into v2.10.1 ?

**Comment by mitchellmaler** [comment]
@trinaths We have successfully upgraded to 2.10.1 now with all the fixes in place. Thanks!
